packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp

```diff
@@ -679,16 +813,37 @@ static EdgeInsets calculateOverflowInset(
   return overflowInset;
 }
 
+void YogaLayoutableShadowNode::setContentsLayoutMetrics(LayoutContext layoutContext) {
+  if (layoutMetrics_ != EmptyLayoutMetrics) {
+    setLayoutMetrics(EmptyLayoutMetrics);
+  }
+
+  for (const auto &child : getChildren()) {
+    if (const auto& layoutableChild = std::dynamic_pointer_cast<const YogaLayoutableShadowNode>(child)) {
+      if (layoutableChild->hasDisplayContentsStyle()) {
+        auto& contentsNode = shadowNodeFromContext(&layoutableChild->yogaNode_);
+        contentsNode.setContentsLayoutMetrics(layoutContext);
+      }
+    }
+  }
+}
+
 void YogaLayoutableShadowNode::layout(LayoutContext layoutContext) {
   // Reading data from a dirtied node does not make sense.
   react_native_assert(!yogaNode_.isDirty());
 
-  for (auto childYogaNode : yogaNode_.getChildren()) {
+  for (auto child : yogaLayoutableChildren_) {
```

Comment:
nit: don't copy shared_ptr by value for each iteration here (previous version was copying raw ptr)

Reply:
Done in https://github.com/facebook/react-native/pull/46584/commits/bc1984f5c3ddc2a529c842377a2dbecb7c679dab